deployment/ccip/changeset/v1_6/cs_chain_contracts.go

```diff
@@ -1265,8 +1273,44 @@ func (cfg UpdateRouterRampsConfig) Validate(e deployment.Environment, state chan
 					return err
 				}
 			} else {
-				if err := commoncs.ValidateOwnership(e.GetContext(), cfg.MCMS != nil, e.Chains[chainSel].DeployerKey.From, chainState.Timelock.Address(), chainState.Router); err != nil {
-					return err
+				// If we are activating ramps on the main router, we should validate two things:
+				//   1. All expected CCIP contracts exist on the chain.
+				//   2. All contracts have the expected owner.
+				// That way, if cfg.MCMS exists, we ensure that every contract is owned by MCMS.
+				ownedContracts := map[string]commoncs.Ownable{
+					"router":             chainState.Router,
+					"feeQuoter":          chainState.FeeQuoter,
+					"offRamp":            chainState.OffRamp,
+					"onRamp":             chainState.OnRamp,
+					"nonceManager":       chainState.NonceManager,
+					"rmnRemote":          chainState.RMNRemote,
+					"rmnProxy":           chainState.RMNProxy,
+					"tokenAdminRegistry": chainState.TokenAdminRegistry,
+				}
+				var wg sync.WaitGroup
+				errs := make(chan error, len(ownedContracts))
```

Comment:
nit: easier implementation with `errgroup.Group{}`

Reply:
@AnieeG errgroup doesn't fit here because once it encounters a non-nil error, it cancels the derived context and returns said error. I want to aggregate **all** errors so that the user is aware of all the contracts that they have to transfer to timelock.

https://github.com/golang/go/issues/23595
https://pkg.go.dev/golang.org/x/sync/errgroup#Group.Go